create-handle/src/test/java/no/sikt/nva/handle/CreateHandleHandlerTest.java

```diff
@@ -21,8 +75,136 @@ public void tearDown() {
+        var appender = LogUtils.getTestingAppenderForRootLogger();
+        handler.handleRequest(inputStream, outputStream, context);
+        var response = GatewayResponse.fromOutputStream(outputStream, CreateHandleResponse.class);
+        assertThat(appender.getMessages(), containsString(String.format(CREATED_HANDLE_FOR_URI,
+                createHandleFromHandleId(CREATED_HANDLE_ID), uri)));
+        assertThat(response.getStatusCode(), is(equalTo(HttpURLConnection.HTTP_CREATED)));
+        assertThat(response.getBodyObject(CreateHandleResponse.class).getHandle(),
+                is(equalTo(createHandleFromHandleId(CREATED_HANDLE_ID))));
+    }
+
+    @Test
+    void createHandleRequestReturnsExistingHandleForValidUriThatAlreadyHasHandle() throws
+            IOException, SQLException {
+        var uri = randomUri();
+        var inputStream = createCreateHandleRequest(uri);
+        mockHandleDatabaseCreateHandle(true, false, false);
+        var appender = LogUtils.getTestingAppenderForRootLogger();
+        handler.handleRequest(inputStream, outputStream, context);
+        var response = GatewayResponse.fromOutputStream(outputStream, CreateHandleResponse.class);
+        assertThat(appender.getMessages(), containsString(String.format(REUSED_EXISTING_HANDLE_FOR_URI,
+                createHandleFromHandleId(EXISTING_HANDLE_ID), uri)));
+        assertThat(response.getStatusCode(), is(equalTo(HttpURLConnection.HTTP_CREATED)));
+        assertThat(response.getBodyObject(CreateHandleResponse.class).getHandle(),
+                is(equalTo(createHandleFromHandleId(EXISTING_HANDLE_ID))));
+    }
+
+    @Test
+    void createHandleRequestReturnsBadRequestResponseWhenUriIsNull() throws IOException {
+        var inputStream = createCreateHandleRequest(null);
+        handler.handleRequest(inputStream, outputStream, context);
+        var response = GatewayResponse.fromOutputStream(outputStream, Problem.class);
+        assertThat(response.getStatusCode(), is(equalTo(HttpURLConnection.HTTP_BAD_REQUEST)));
+        assertThat(response.getBody(), containsString(NULL_URI_ERROR));
+    }
+
+    @Test
+    void createHandleRequestReturnsBadGatewayAndLogsErrorWhenNotAbleToCreateHandleId() throws IOException,
+            SQLException {
+        var uri = randomUri();
+        var inputStream = createCreateHandleRequest(uri);
+        mockHandleDatabaseCreateHandle(false, false, false);
+        var appender = LogUtils.getTestingAppenderForRootLogger();
+        handler.handleRequest(inputStream, outputStream, context);
+        var response = GatewayResponse.fromOutputStream(outputStream, Problem.class);
+        assertThat(appender.getMessages(), containsString(String.format(ERROR_CREATING_HANDLE_FOR_URI, uri)));
+        assertThat(response.getStatusCode(), is(equalTo(HttpURLConnection.HTTP_BAD_GATEWAY)));
+    }
+
+    @Test
+    void createHandleRequestReturnsBadGatewayAndLogsErrorAndRollbackDatabaseTransactionWhenNotAbleToCreateHandle()
```

Comment:
It seems we are connected to a relational database. Do we have to rollback manually the transaction? Doesn't it happen automatically?